designs/helm-chart-autogenerate-plugin.md

```diff
@@ -0,0 +1,271 @@
+
+### Non-Goals
+
+- Converting any Kustomize configuration to Helm Charts like [helmify](https://github.com/arttor/helmify) does.
+- Support the deprecated plugins. This option should be supported from go/v4 and kustomize/v2
+- Introduce support for Helm in addition to Kustomize, or replace Kustomize with Helm entirely, similar to the approach taken by Operator-SDK, thereby allowing users to utilize Helm Charts to build their Operators.
+
+## Proposal
+
+The new optional plugin will create a new directory, `deploy/helm-chart`, where the scaffolding of the project should be constructed as follows:
+//TODO: We need a better definition 
+// We must create a Helm Chart with the default Kubebuilder scaffold to add here
+```shell
+❯ tree deploy/helm-chart
+helm-chart
+    ├── Chart.yaml
+    ├── crds
+    │   ├── <CRDs YAML files generate under config/crds/>
+    ├── templates
+    │   ├── deployment.yaml 
+    │   ├── _helpers.tpl
+    │   ├── role_binding.yaml
+    │   ├── role.yaml 
+    │   ├── <all under config/rbac less the edit and view roles which are helpers for admins>
+    └── values.yaml
+```
+
+To reduce the user adaptation, the command
+leverage the existing kustomize default directory and create the similar layout of the helm chart from it. The generated `values.yaml` is as follows:
+
+```yaml
+# global configurations
+nameOverride: ""
+
+fullnameOverride: ""
+
+## CRD configuration under the `config/crd` directory
+crd:
+  ## Whether the `resources` field contains `crd` in the `config/default/kustomization.yaml` file
+  create: false
+
+## RBAC configuration under the `config/rbac` directory
+rbac:
+  ## Whether the `resources` field contains `rbac` in the `config/default/kustomization.yaml` file
+  create: true
+  serviceAccountName: "controller-manager"
+
+
+## Manager configuration referent to `config/manager/manager.yaml`
+manager:
+  ## The Security Context of the manager Pod.
+  ## More info: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod
+  podSecurityContext:
+    runAsNonRoot: true
+
+  ## Configure the kubeRbacProxy container.
+  kubeRbacProxy:
+    image:
+     registry: gcr.io
+     repository: "kubebuilder/kube-rbac-proxy"
+     tag: v0.13.0
+     pullPolicy: IfNotPresent
+
+  ## Configure the resources.
+  ## More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/ 
+  resources:
+    limits:
+      cpu: 500m
+      memory: 128Mi
+    requests:
+      cpu: 5m
+      memory: 64Mi
+
+  ## Configure the controller container.
+  controller:
+    image:
+      registry: docker.io
+      repository: ""
+      tag: latest
+      pullPolicy: IfNotPresent
+
+     ## The environment variables of the controller container.
+     ## More info: https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/
+  	env: []
```

Comment:
Do we have our default Kubebuilder Operator layout or any plugin?
If not, then I think we need to remove it.
If yes, we need to add a comment here to justify why/when we need/use this one.

Or goal is just package the Operator in HelmChart
Any extra thing or customizations that might be done on top shows out of our scope.

Reply:
> Do we have our default Kubebuilder Operator layout or any plugin?

You mean that here we should only take inputs we already have in kustomize, for example? Inputs that kubebuilder currently already knows how to work with.

I think that makes sense, just clarifying.